cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -118,6 +118,60 @@
   protected void configure() {
     setMainClass(SparkPersistRunRecordMain.class);
     addHandlers(new ReportSparkHandler());
+    String version = getSparkVersion();
+    if (!isSparkVersionSupported(version)) {
+      throw new RuntimeException(
+        String.format("Spark Version %s is not supported, Please upgrade to Spark version 2.1 or later", version));
+    }
+  }
+
+  /**
+   * Spark version returned using SparkContext#version() is set by org.apache.spark.SparkBuildInfo in a similar manner
+   * by loading the spark-version-info.properties file and getting the property version.
+   * @return version string, if unable to load the file due to any error, return null
+   */
+  @Nullable
+  private String getSparkVersion() {
+    ClassLoader classLoader = org.apache.spark.SparkContext.class.getClassLoader();
+    InputStream inputStream = classLoader.getResourceAsStream("spark-version-info.properties");
+    if (inputStream == null) {
+      return null;
+    }
+    Properties properties = new Properties();
+    try {
+      properties.load(inputStream);
+    } catch (IOException e) {
+      return null;
+    }
+    String version = properties.getProperty("version");
+    return version;
+  }
+
+  /**
+   * ReportGeneration is supported only from Spark version 2.1 currrently, we return false for versions lower than 2.1
+   */
+  private boolean isSparkVersionSupported(String version) {
+    LOG.info("Spark version is {}", version);
+    // if version cannot be loaded assume its supported
+    if (version == null || version.isEmpty()) {
```

Comment:
Perhaps better to assume not supported by default.

Reply:
I think it's ok if we let this pass. If it's spark1, the UI won't even let you enable the app, and we've checked that this works for spark2.0. If they have changed how version is done in later versions, we should be ok with that.